src/BuildDacpac/Extensions.cs

```diff
@@ -9,15 +9,38 @@ public static class Extensions
     {
         public static void AddReference(this TSqlModel model, string referencePath)
         {
-            var service = model.GetType().GetField("_service", BindingFlags.NonPublic | BindingFlags.Instance).GetValue(model);
-            var dataSchemaModel = service.GetType().GetProperty("DataSchemaModel", BindingFlags.NonPublic | BindingFlags.Instance).GetValue(service);
+            var dataSchemaModel = CreateDataSchemaModel(model);
 
             var customData = Activator.CreateInstance(Type.GetType("Microsoft.Data.Tools.Schema.SchemaModel.CustomSchemaData, Microsoft.Data.Tools.Schema.Sql"), "Reference", "SqlSchema");
             var setMetadataMethod = customData.GetType().GetMethod("SetMetadata", BindingFlags.Public | BindingFlags.Instance);
             setMetadataMethod.Invoke(customData, new object[] { "FileName", referencePath });
             setMetadataMethod.Invoke(customData, new object[] { "LogicalName", Path.GetFileName(referencePath) });
             setMetadataMethod.Invoke(customData, new object[] { "SuppressMissingDependenciesErrors", "False" });
 
+            InvokeAddMethod(dataSchemaModel, customData);
+        }
+
+
+        public static void AddSqlCmdVariable(this TSqlModel model, string variableName)
+        {
+            var dataSchemaModel = CreateDataSchemaModel(model);
+
+            var customData = Activator.CreateInstance(Type.GetType("Microsoft.Data.Tools.Schema.SchemaModel.CustomSchemaData, Microsoft.Data.Tools.Schema.Sql"), "SqlCmdVariables", "SqlCmdVariable");
+            var setMetadataMethod = customData.GetType().GetMethod("SetMetadata", BindingFlags.Public | BindingFlags.Instance);
+            setMetadataMethod.Invoke(customData, new object[] { variableName, string.Empty });
+
+            InvokeAddMethod(dataSchemaModel, customData);
+        }
+
+        private static object CreateDataSchemaModel(TSqlModel model)
```

Comment:
Perhaps this would be more appropriately named `GetDataSchemaModel` since it doesn't really create anything.